src/envs/behavior_options.py

```diff
@@ -0,0 +1,976 @@
+        robot=env.robots[0],
+        obj_in_hand=obj_in_hand,
+        end_conf=[
+            x,
+            y,
+            z + 0.2,
+            0,
+            np.pi * 7 / 6,
+            0,
+        ],
+        hand_limits=((minx, miny, minz), (maxx, maxy, maxz)),
+        obstacles=obstacles,
+        rng=rng,
+    )
+    p.restoreState(original_state)
+    p.removeState(original_state)
+
+    # NOTE: This below line is *VERY* important after the
+    # pybullet state is restored. The hands keep an internal
+    # track of their state, and if we don't reset this
+    # state to mirror the actual pybullet state, the hand will
+    # think it's elsewhere and update incorrectly accordingly
+    env.robots[0].parts["right_hand"].set_position(
+        env.robots[0].parts["right_hand"].get_position())
+    env.robots[0].parts["left_hand"].set_position(
+        env.robots[0].parts["left_hand"].get_position())
+
+    return plan
+
+
+# NOTE: we need a type:ignore because we can't specify the type of obj because
+# it would be dependent on an optional igibson import which won't get imported
+# on CI.
+def place_ontop_obj_pos_sampler(  # type: ignore
+        obj,
+        rng: Generator = np.random.default_rng(23),
+) -> Array:
+    """Sampler for placeOnTop option."""
+    # objA is the object the robot is currently holding, and objB
+    # is the surface that it must place onto.
+    # The BEHAVIOR NSRT's are designed such that objA is the 0th
+    # argument, and objB is the last.
+    objA = obj[0]
+    objB = obj[-1]
+
+    params = _ON_TOP_RAY_CASTING_SAMPLING_PARAMS
+    aabb = get_aabb(objA.get_body_id())
+    aabb_extent = get_aabb_extent(aabb)
+
+    _ = rng.random()
+    sampling_results = sampling_utils.sample_cuboid_on_object(
+        objB,
+        num_samples=1,
+        cuboid_dimensions=aabb_extent,
+        axis_probabilities=[0, 0, 1],
+        refuse_downwards=True,
+        **params,
+    )
+
+    if sampling_results[0] is None or sampling_results[0][0] is None:
+        # If sampling fails, returns a particular constant set of params
+        return np.array([0.0, 0.0, 0.5])
+
+    rnd_params = np.subtract(sampling_results[0][0], objB.get_position())
+
+    # NOTE: The below block returns the orientation of the object in addition
+    # to the rnd_params, which is useful for the option model implementation.
+    # return_orn can be passed in as an argument to the function
+    # if return_orn:
+    #     return rnd_params, sampling_results[0][2]
+
+    return rnd_params
+
+
+# NOTE: we need a type:ignore because we can't specify the type of env because
+# it would be a igibson.envs.behavior_env.BehaviorEnv, which is an optional
+# import and won't get imported on CI
+def place_ontop_obj_pos(  # type: ignore # pylint: disable=inconsistent-return-statements
+        env,
+        obj,
+        place_rel_pos: Array,
+        place_orn: Optional[Array] = None,
+        option_model: bool = False,
```

Comment:
the option model code is not yet ready for prime time, right? can you remove it until it is?